plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt

```diff
@@ -145,6 +150,19 @@ abstract class LoginBrowser(
     }
 
     open fun loginIdC(url: String, region: AwsRegion, scopes: List<String>) {
+        // assumes scopes contains either Q or non-Q permissions but not both
+        val isReAuth = if (scopes.toSet().intersect(Q_SCOPES.toSet()).isNotEmpty()) {
+            ToolkitConnectionManager.getInstance(project).activeConnectionForFeature(QConnection.getInstance()) != null
```

Comment:
Should we also compare the start url and region to the active connection's start url and region, cause if they are different then technically it's a new auth?

Reply:
Can we also move the isReauth logic to a new function?

Reply:
> Should we also compare the start url and region to the active connection's start url and region, cause if they are different then technically it's a new auth?

Gonna check in with analytics to see if this is considered new auth. There could be a world where the following can be considered a continuous string of reauths:

User relogs with acc A -> A credentials expire -> user relogs with acc B -> B credentials expire -> user relogs with acc C -> C credentials expire -> ...